Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -298,35 +321,36 @@ void OpenALStream::SoundLoop()
       }
     }
 
-    unsigned int nSamples = soundTouch.receiveSamples(sampleBuffer.data(), OAL_MAX_SAMPLES * numBuffers);
+	// We want SoundTouch to return already processed samples all at once
+    unsigned int n_samples = soundTouch.receiveSamples(sample_buffer.data(), soundTouch.numSamples());
 
-    if (nSamples <= minSamples)
+    if (n_samples < rendered_samples)
       continue;
 
     if (surround_capable)
     {
-      std::vector<float> dpl2(OAL_MAX_SAMPLES * OAL_MAX_BUFFERS * SURROUND_CHANNELS);
-      DPL2Decode(sampleBuffer.data(), nSamples, dpl2.data());
+      std::vector<float> dpl2(n_samples * SURROUND_CHANNELS);
```

Comment:
Sometimes n_samples > rendered_samples when using SoundTouch (I am not asking for SoundTouch a fixed number of samples, I am asking for all samples it has available). Also, as I said, if I change soundTouch.numSamples() back to rendered_samples the latency increases (and it is very noticeable)

I have a hunch that SoundTouch only likes to "send" samples which are a multiple of 576...